File: frontend/src/api/wireguard/request/get_peers.ts
import { api } from 'boot/axios';
import API from 'api/wireguard/api';
import Peer from 'api/wireguard/data/peer';

export interface IGetPeersResponse extends Response {
  peers: Peer[];
}

export default function getPeers() {
  return api.get<IGetPeersResponse>(API.PEERS);
}

File: backend/wireguard/request/get_peers.py
from dataclasses import dataclass, field

from api.base_request import BaseRequest
from api.base_response import BaseResponse

from wireguard.data.peer import Peer


@dataclass
class GetPeersRequest(BaseRequest):
    pass


@dataclass
class GetPeersResponse(BaseResponse):
    peers: list[Peer] = field(default_factory=list)
